main.go

```diff
@@ -183,8 +186,8 @@ func main() {
 
 	router := route.New()
 
-	RegisterWeb(router)
-	api.Register(router.WithPrefix("/api"))
+	RegisterWeb(router.WithPrefix(amURL.Path))
+	api.Register(router.WithPrefix(amURL.Path + "/api"))
```

Comment:
path.Join does not work well for URLs in general, I usually decide against using it.

https://play.golang.org/p/xGEOg3Gw7p